### ISSUE TITLE
Block toolbar DOM order and visual order mismatch in the responsive view

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#14419]
Splitting this out from #9739 

In the responsive view, the block toolbar changes position and it's placed _after_ the block together with the block "movers" and the Add block button:

<img width="421" alt="Screenshot 2019-03-13 at 18 39 17" src="https://user-images.githubusercontent.com/1682452/54301506-61007200-45bf-11e9-999b-1f155620e64c.png">

However, to my understanding there's some DOM repositioning but it's mainly a visual repositioning. The DOM order is mostly the same as in the desktop view. This way, visual order and DOM order mismatch and don't make any sense.

Worth reminding [many screen reader users use a keyboard when using their mobile devices](https://webaim.org/projects/screenreadersurvey7/#mobilekeyboard) so keyboard operability is relevant also in the mobile view.

It's also a WCAG failure but, regardless, it makes operating with a keyboard nearly impossible and should be fixed. If a reordering is desired, it should change also the DOM order to match the visual order.
